lib/stdlib/test/erl_scan_SUITE.erl

```diff
@@ -430,25 +441,119 @@ floats() ->
          fun({S, I}) ->
                  test_string(S, [{float, {1, 1}, I}])
          end, UnderscoreSamples),
-    UnderscoreErrors =
-        ["123_.456",
-         "123._456",
-         "123.456_",
-         "123._",
-         "1._23e10",
+    FloatErrors =
+        ["123.456_",
+         "1.23_e10",
          "1.23e_10",
-         "1.23e10_"],
+         "1.23e10_",
+         "123.45_e6",
+         "123.45a12",
+         "123.45e23a12",
+         "1.e2",
+         "12._34",
+         "123.a4"
+        ],
     lists:foreach(
       fun(S) ->
               case erl_scan:string(S) of
```

Comment:
To improve code coverage, I suggest updating the existing helper function `erl_scan_string/3` like this:

```
erl_scan_string(String, StartLocation, Options) ->
    case erl_scan:string(String, StartLocation, Options) of
        {ok, Tokens, EndLocation} ->
            {ok, unopaque_tokens(Tokens), EndLocation};
        {error,{_,Mod,Reason},_}=Error ->
            Mod:format_error(Reason),
            Error
    end.
```

Then replace all calls to `erl_scan:string/1` in this PR with `erl_scan_string/1`.
```suggestion
              case erl_scan_string(S) of
```